### PR TITLE
Test infrastructure readme update

### PR DESCRIPTION
**Pull request by hadarai**
This PR updates our test infrastructure readme with my knowledge I gained during development of pipe tests

### CODE REVIEW

**Review by pwit81 — APPROVED** [review]

File: wiki/docs/tests.md [hunk 1]
```diff
@@ -31,36 +30,99 @@ listed below:
   `repeat` specifies the number of times each test will be run. For example,
   `test=test1,test2 repeat=4` will run `test1` 4 times and then run `test2` 4 times.
 
-### Programming tests
+## Programming tests
```
Comment (wiklam) [inline wiki/docs/tests.md:33]:
```suggestion
## Tests implementation
```
I am not sure about this one

File: wiki/docs/tests.md [hunk 2]
```diff
@@ -31,36 +30,99 @@ listed below:
...
   status `{exit status}`.
 
-One also needs to add a line `CHECKRUN_TEST({name})` in `/user/utest/main.c`.
+One also needs to add a line
+
+* `CHECKRUN_TEST({name})` in `/bin/utest/main.c`,
+* `int test_{name}(void);` in `/bin/utest/utest.h`,
+* `${filename}.c` in `bin/utest/Makefile`.
+
+#### Creating tests
```
Comment (wiklam) [inline wiki/docs/tests.md:70]:
```suggestion
### Creating tests
```

File: wiki/docs/tests.md [hunk 3]
```diff
@@ -31,36 +30,99 @@ listed below:
...
+
+Tests are supposed to return `0` or create kernel panic during execution,
+that can be achieved by usage of `assert`. That implies that one should change
+parts of its code that may look like this
```
Comment (wiklam) [inline wiki/docs/tests.md:74]:
```suggestion
that can be achieved by the usage of `assert`. That implies that the test
shouldn't look like the code bellow
```

File: wiki/docs/tests.md [hunk 4]
```diff
@@ -31,36 +30,99 @@ listed below:
...
+    /* child */
+    ...
+
+default:               
+    /* parent */
+    ...
+}
+```
+
+to more like
```
Comment (wiklam) [inline wiki/docs/tests.md:94]:
```suggestion
but more like this one
```

**Review by wiklam — APPROVED** [review]
The rest seems fine, good job!

**Pull request merged**